### PR TITLE
Add CI job to run spec-parser and upload its output

### PR DESCRIPTION
**Pull request by maxhbr**
- this clones and runs https://github.com/spdx/spec-parser/
- the generated files are uploaded as artifacts to the CI job

### CODE REVIEW

**Comment by maxhbr** [comment]
An example CI run can be found here: https://github.com/maxhbr/spdx-3-model/actions/runs/4525612756. There you can also download the zip containing all artifacts

**Comment by zvr** [comment]
Thanks, @maxhbr, this is great.

Can we have the generated output appearing in a `generated`branch?

**Comment by maxhbr** [comment]
sure, but I would leave that for a follow-up PR

File: .github/workflows/spec-parser.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: get spec-parser
+        run: |
+          git clone https://github.com/spdx/spec-parser/
```
Comment (zvr) [inline .github/workflows/spec-parser.yml:18]: `--depth=1 --single-branch --branch=main` is what I'm usually adding (not that it matters much in this case)

**Pull request merged**